cli/azd/cmd/container.go

```diff
@@ -602,4 +605,95 @@ func registerCommonDependencies(container *ioc.NestedContainer) {
 	registerAction[*provisionAction](container, "azd-provision-action")
 	registerAction[*downAction](container, "azd-down-action")
 	registerAction[*configShowAction](container, "azd-config-show-action")
+
+	container.RegisterSingleton(func(
+		ctx context.Context,
+		userConfigManager config.UserConfigManager,
+		lazyProjectConfig *lazy.Lazy[*project.ProjectConfig],
+		lazyAzdContext *lazy.Lazy[*azdcontext.AzdContext],
+		lazyLocalEnvStore *lazy.Lazy[environment.LocalDataStore],
+	) (*cloud.Cloud, error) {
+
+		// 1. Check config (~/.azure/config.json) set by azd config set (cloud node)
+		// 2. Check project config (azure.yaml)
+		// 3. Check .azure/<environment>/config.json
+
+		// Default configuration
+		var cloudConfig = &cloud.Config{Name: cloud.AzurePublicName}
+		var suggestion func() string
+
+		// User Configuration
+		if azdConfig, err := userConfigManager.Load(); err == nil {
+			cloudConfigNode, exists := azdConfig.Get(cloud.ConfigPath)
+			if exists {
+				value, err := cloud.ParseCloudConfig(cloudConfigNode)
+				if err == nil {
+					cloudConfig = value
+
+					// In the event of an error set the suggestion for updating the cloud configuration
+					suggestion = func() string {
+						return "Set the cloud configuration using 'azd config set cloud.name <name>'. "
+					}
+				}
+
+			}
+		}
+
+		// Project Configuration
+		projConfig, _ := lazyProjectConfig.GetValue()
```

Comment:
You likely should check for an error here in the event there is no project / azure.yaml file.